python/graphstorm/trainer/glem_np_trainer.py

```diff
@@ -256,41 +256,48 @@ def _prepare_batch(input_nodes, seeds, blocks, is_labeled=True):
                              get_rank(), epoch, i,  loss.item(), time.time() - batch_tic)
 
             val_score = None
-            if self.evaluator is not None and \
-                self.evaluator.do_eval(total_steps, epoch_end=False) and \
-                val_loader is not None:
+            if self.can_do_model_eval(val_loader) and self.evaluator.do_eval(total_steps):
                 val_score = self.eval(model.module, val_loader, test_loader,
-                                        use_mini_batch_infer, total_steps, return_proba=False)
+                                      use_mini_batch_infer, total_steps, return_proba=False)
 
                 if self.evaluator.do_early_stop(val_score):
                     self.early_stop = True
 
+            # In every save_model_frequency iterations, check to save the top k models.
+            # If has validation score, will save the best top k. If no validation, will
+            # either save the last k model or all models depends on the setting of top k.
             if save_model_frequency > 0 and \
                 total_steps % save_model_frequency == 0 and \
                 total_steps != 0:
-                if self.evaluator is None or val_score is not None:
+                if self.can_do_model_eval(val_loader):
```

Comment:
Can we find a better naming for can_do_model_eval?

Reply:
like `do_model_eval`?

Reply:
Per discussion, changed to `can_do_validation()`.